### PR TITLE
[MH2] Implement Altar of the Goyf, Blazing Rootwalla, Clattering Augur, and Damn

### PR DESCRIPTION
**Pull request by jmharmon**

### CODE REVIEW

File: Mage.Sets/src/mage/cards/a/AltarOfTheGoyf.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    public AltarOfTheGoyf copy() {
+        return new AltarOfTheGoyf(this);
+    }
+}
+
+class AltarOfTheGoyfAbility extends TriggeredAbilityImpl{
+
+    public AltarOfTheGoyfAbility() {
+        super(Zone.BATTLEFIELD, new BoostTargetEffect(
+                CardTypesInGraveyardCount.instance, CardTypesInGraveyardCount.instance, Duration.EndOfTurn, true), false);
```
Comment (theelk801) [inline Mage.Sets/src/mage/cards/a/AltarOfTheGoyf.java:62]: this needs to count the card types in all graveyards, not just the controller's

Comment (theelk801) [inline Mage.Sets/src/mage/cards/a/AltarOfTheGoyf.java:62]: also needs `CardTypesInGraveyardHint.ALL`

File: Mage.Sets/src/mage/cards/a/AltarOfTheGoyf.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    public boolean checkEventType(GameEvent event, Game game) {
+        return event.getType() == GameEvent.EventType.DECLARED_ATTACKERS;
+    }
+
+    @Override
+    public boolean checkTrigger(GameEvent event, Game game) {
+        if (game.isActivePlayer(this.controllerId)) {
+            if (game.getCombat().attacksAlone()) {
+                for (Effect effect : this.getEffects()) {
+                    effect.setTargetPointer(new FixedTarget(game.getCombat().getAttackers().get(0)));
```
Comment (theelk801) [inline Mage.Sets/src/mage/cards/a/AltarOfTheGoyf.java:84]: use `this.getEffects().setTargetPointer(new FixedTarget(game.getCombat().getAttackers().get(0),game))` so you get the zone change counter and don't have to loop through effects

**Review by theelk801 — CHANGES_REQUESTED** [review]

File: Mage.Sets/src/mage/cards/a/AltarOfTheGoyf.java (hunk 1 above)
Comment (jmharmon) [reply Mage.Sets/src/mage/cards/a/AltarOfTheGoyf.java:62]: Done

Comment (jmharmon) [reply Mage.Sets/src/mage/cards/a/AltarOfTheGoyf.java:62]: Done

File: Mage.Sets/src/mage/cards/a/AltarOfTheGoyf.java (hunk 2 above)
Comment (jmharmon) [reply Mage.Sets/src/mage/cards/a/AltarOfTheGoyf.java]: Done

File: Mage.Sets/src/mage/cards/a/AltarOfTheGoyf.java (hunk 1 above)
Comment (jmharmon) [reply Mage.Sets/src/mage/cards/a/AltarOfTheGoyf.java:62]: I'm stuck on this one.

Comment (theelk801) [reply Mage.Sets/src/mage/cards/a/AltarOfTheGoyf.java:62]: where are you stuck with it?

Comment (jmharmon) [reply Mage.Sets/src/mage/cards/a/AltarOfTheGoyf.java:62]: Getting the cards types in all graveyards.

Comment (theelk801) [reply Mage.Sets/src/mage/cards/a/AltarOfTheGoyf.java:62]: you just need to iterate through all players, use `game.getState().getPlayersInRange()`

Comment (jmharmon) [reply Mage.Sets/src/mage/cards/a/AltarOfTheGoyf.java:62]: The only place that seems to fit is the `if (game.isActivePlayer(this.controllerId))` section, but that doesn't work. Trying to look at the code for something like Tarmogoyf doesn't help either.

Comment (weirddan455) [reply Mage.Sets/src/mage/cards/a/AltarOfTheGoyf.java:62]: You need to replace `CardTypesInGraveyardCount.instance` with your own class/enum that implements `DynamicValue`.  Put the code that iterates through the players in the `calculate` method.  Look at the code for `CardTypesInGraveyardCount` as an example to get you started.

**Pull request merged**